src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -72,7 +72,7 @@ def get_responses(self) -> pd.DataFrame:
     def on_collect_metrics(self, event: Event) -> None:
         if self.to_observe(event):
             self.do_observation(event)
```

Comment:
I just thought of something. We can enforce that we maintain the output schema here and throw an exception if `self.responses` doesn't match the schema after `do_observation` is called.

Reply:
Hmm, good idea. I've started thinking a little bit about how we might want to enforce schema throughout this project but this is a good intermediate